torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -416,13 +413,16 @@ class TORCH_API ProcessGroupNCCL : public Backend {
 
   c10::intrusive_ptr<Work> endCoalescing() override;
 
+  // For specifying a composite optype, such as ALLGATHER and REDUCE_SCATTER
+  c10::intrusive_ptr<Work> endCoalescing(OpType optype);
```

Comment:
hmm this one is not an obvious devices->device side effect

Reply:
You are right.
At one point, I was testing the non-even all-gather with this PR. That would translate into coalesced broadcasts. And I found that the flight recorder needs an OpType for naming every work, even for coalesced work. So this change is for being friendlier with flight recorder.